src/server/api/routes/port.router.js

```diff
@@ -0,0 +1,54 @@
+"use strict";
+
+// router setup
+
+const express = require("express");
+const router = express.Router({
+  mergeParams: true
+});
+
+const { authorizeUser } = require("../lib/middleware/auth.middleware");
+
+const ROLES = require("../../constants/roles");
+
+// controllers
+const portController = require("../controllers/port.controller");
+// ENDPOINT: /api/ports/:id :GET
+router.get(
+  "/:id",
+  authorizeUser(ROLES.SUPER_USER, ROLES.ADMIN, ROLES.OPERATOR),
+  function(req, res, next) {
+    portController
+      .getPortById(req.params.id)
+      .then(result => res.json(result))
+      .catch(next);
+  }
+);
+
+// ENDPOINT: /api/ports/ :POST
+router.post(
+  "/",
+  authorizeUser(ROLES.SUPER_USER, ROLES.ADMIN),
+  (req, res, next) => {
+    portController
+      .createPort({
+        body: req.body
+      })
+      .then(result => res.json(result))
+      .catch(next);
+  }
+);
+
+// ENDPOINT: /api/waypoints/:waypoint_id/ports/ :GET
```

Comment:
The URL to this one is actually `/api/ports/:waypoint_id/ports/` since you put it in `ports.router.js` which in `router.js` is mappd to `/ports`.

I realize that this endpoint should probably have been part of the task to do waypoints, because the solution is to move this code to the `waypoints.router.js`, but then it also makes more sense to have `getPortByWaypointId()` in `waypoints.controller.js`.

In the meantime, however, Daniel and I discovered that we could improve the DB schema further by not connecting waypoints and ports, so we will be removing this soon, so you can actually disregard this part of the task altogether. Sorry for the extra work.

Please remove this section.